homeassistant/components/homematicip_cloud/binary_sensor.py

```diff
@@ -136,6 +137,45 @@ async def async_setup_entry(
         async_add_entities(entities)
 
 
+class HomematicipCloudConnectionSensor(HomematicipGenericEntity, BinarySensorEntity):
+    """Representation of the HomematicIP cloud connection sensor."""
+
+    def __init__(self, hap: HomematicipHAP) -> None:
+        """Initialize the cloud connection sensor."""
+        super().__init__(hap, hap.home, "Cloud Connection")
+
+    @property
+    def device_info(self) -> Dict[str, Any]:
+        """Return device specific attributes."""
+        # Adds a sensor to the existing HAP device
+        return {
+            "identifiers": {
+                # Serial numbers of Homematic IP device
+                (HMIPC_DOMAIN, self._device.id)
+            }
+        }
+
+    @property
+    def icon(self) -> str:
+        """Return the icon of the access point entity."""
+        return "mdi:access-point-network"
```

Comment:
when you use device class, the icon is picked/handled by frontend.

https://github.com/home-assistant/frontend/blob/82f80db558ceaab1d1cbe54fbfe2e2165e7cacd5/src/common/entity/binary_sensor_icon.ts#L12

Reply:
I removed the device_class, because 'connectivity' currently returns mdi:alert.